src/components/Router/HistoryRouter.tsx

```diff
@@ -87,6 +106,33 @@ export const HistoryRouter = (props: HistoryRouterProps) => {
     })
   }
 
+  const [passiveSignalsTracker, setPassiveSignalsTracker] = useState<
```

Comment:
I think it would be better to abstract the code into it's own hook, like `usePassiveSignalsTracker()`. Now the code it polluting the router.

Reply:
Make sense, combining with the above suggestion, we can encapsulate the consumption of `useSdkConfigurationService` from `HistoryRouter`

Reply:
Checked out as `usePassiveSignals` custom hook